packages/sdk/src/contracts/SponsorshipFactory.ts

```diff
@@ -0,0 +1,55 @@
+import { SponsorshipFactoryABI } from '@streamr/network-contracts'
+import { EthereumAddress, StreamID, toEthereumAddress, toStreamID } from '@streamr/utils'
+import { Interface } from 'ethers'
+import { Lifecycle, inject, scoped } from 'tsyringe'
+import { ConfigInjectionToken, StrictStreamrClientConfig } from '../Config'
+import { StreamrClientEventEmitter } from '../events'
+import { LoggerFactory } from '../utils/LoggerFactory'
+import { ChainEventPoller } from './ChainEventPoller'
+import { initContractEventGateway } from './contract'
+
+export interface SponsorshipCreatedEvent {
+    readonly sponsorshipContractAddress: EthereumAddress
+    readonly streamId: StreamID
+    readonly metadata: string
```

Comment:
This could be similar to `StreamMetadata`:

```suggestion
    readonly metadata: Record<string, unknown>
```

Reply:
I'd put it future improvements now, since we don't rely on metadata atm